src/Symfony/Component/Config/Definition/Builder/TreeBuilder.php

```diff
@@ -43,9 +44,13 @@ public function __construct(string $name = null, string $type = 'array', NodeBui
      * @return ArrayNodeDefinition|NodeDefinition The root node (as an ArrayNodeDefinition when the type is 'array')
      *
      * @throws \RuntimeException When the node type is not supported
+     *
+     * @deprecated since Symfony 4.3, pass the root name to the constructor instead
      */
     public function root($name, $type = 'array', NodeBuilder $builder = null)
     {
+        @trigger_error(sprintf('The "%s()" method (called for "%s") is deprecated since Symfony 4.3, pass the root name to the constructor instead.', __METHOD__, $name), E_USER_DEPRECATED);
```

Comment:
maybe `called for the "%s" configuration` (something that hints what %s is)?

Reply:
And remove the brackets I think.